components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/HttpRequestOperation.java

```diff
@@ -154,16 +156,17 @@ private static boolean requestIsOngoing(RequestBodyChunkSubscriber bodyChunkSubs
                     });
         }
 
-        return observable.doOnTerminate(() -> {
-            if (nettyConnection.isConnected()) {
-                removeProxyBridgeHandlers(nettyConnection);
-                if (requestIsOngoing(requestRequestBodyChunkSubscriber.get())) {
-                    LOGGER.warn("Origin responded too quickly to an ongoing request, or it was cancelled. Connection={}, Request={}.",
-                            new Object[]{nettyConnection.channel(), this.request});
-                    nettyConnection.close();
-                }
-            }
-        }).map(response -> response.newBuilder().build());
+        return observable.map(response ->
+                        Requests.doFinally(response, cause -> {
```

Comment:
In all honesty the `ResponseEventListener` class got implemented after this change. However, it is not necessary to roll out a full response event listener in this instance because all we do here is to remove Netty handlers from the channel. We don't care if it is because cancellation or because of end of the stream. We just need an "end-of-stream" signal that triggers the channel clean up.

Additionally, the `ResponseEventListener` coordinates between reactive signals from a Styx `Eventual` and a `ByteStream`. For example it collapses response `Eventual` cancellation with `ByteStream` cancellation. It provides a single `whenCancelled` interface for the event listener regardless if the response was cancelled via `Eventual` or `ByteStream`.  It also deals with some interdependencies and signal ordering issues involved in response cancellation.

So because response event listener adds complexity by executing code that is not relevant for this use case, it is better to use a simpler and more straightforward `doFinally` hook to trigger a cleanup.